pkg/util/util.go

```diff
@@ -173,23 +174,48 @@ func GetLocalEndpoint(submSpec types.SubmarinerSpecification, backendConfig map[
 	return endpoint, nil
 }
 
-func getLabelPort(gwNode *v1.Node, labelName string) (*int32, error) {
-	var natDiscoveryPort *int32
+func getBackendConfig(nodeObj *v1.Node) (map[string]string, error) {
+	validConfigs := stringset.New(subv1.ValidBackendConfigs...)
+	backendConfig := map[string]string{}
+
+	for label, value := range nodeObj.Labels {
+		if strings.HasPrefix(label, node.BackendConfigPrefix) {
+			config := label[len(node.BackendConfigPrefix):]
+			if !validConfigs.Contains(config) {
+				return backendConfig, errors.Errorf("unknown backend config from node %q label %q",
+					nodeObj.Name, label)
+			}
 
+			backendConfig[config] = value
+		}
+	}
```

Comment:
Is there a use case where a user would want different backend configs for different gateway nodes? If not then they'd have to maintain these labels on every gateway node. We gave a [proposal](https://github.com/submariner-io/enhancements/pull/4) for using policies to configure cable driver configs. It seems this may conflict with proposal. I don't want us stuck with supporting/migrating something legacy if we go with that proposal.

Reply:
Yes we need different backend configs per gateway node, it's that use case what I'm trying to resolve. 

But limited ones, we could be very restrictive in the filters of what's accepted as a configuration specific per-gateway, for example:

* Ports
* "I prefer to be server" (that could be cluster-wide in most cases, but I can imagine situations where it wouldn't), we certainly need this to fix connectivity & NAT completely for production.


So this won't become legacy, unless the other proposal has a way to express that, but I believe this local configuration detail is probably more reasonable to do in the cluster itself.

Reply:
Based on this conversation I have updated this here: https://github.com/submariner-io/submariner/pull/1220/files#diff-c52b44daa2be3d01c651d0fe4dc5821b8da017dfabd4cc60830911b37445b0bfR89